bin/moonpick

```diff
@@ -20,45 +20,79 @@ Usage: moonpick [opts] <file>, [file2, ..]
 
 Where [opts] can be the following:
 
--v Shows moonpick version
--h Shows this help
+-v                Shows moonpick version
+-h                Shows this help
+--filename file   Filename used for lint_config with stdin
+
+--                Read from stdin, must be last argument
 ]]
     os.exit(1)
   end
 
   if #args == 0 then bail() end
-
-  for _, a in ipairs(args) do
-    if a:match('^%-') then
+  local opts = {}
+  for i, a in ipairs(args) do
+    if a == "--" then
+      opts.stdin = true
+      break
+    elseif a:match('^%-') then
       if a == '-v' or a == '--version' then
         print("Moonpick version " .. VERSION)
         os.exit(0)
+      elseif a == '--filename' then
+          local file = args[i + 1]
+          if not file then
+            print("Missing value for option --filename")
+            os.exit(0)
+          end
+          opts.filename = args[i + 1]
       else
         bail()
       end
     end
   end
+  return opts
 end
 
 local args = {...}
-validate_usage(args)
+local opts = validate_usage(args)
 
 local errors = 0
-
-for i = 1, #args do
-  local file = args[i]
-  local inspections, err = moonpick.lint_file(file)
+local function lint(linter, filename)
+  local inspections, err = linter()
   if not inspections then
     errors = errors + 1
-    print(file .. '\n')
+    if filename then
+      print(filename .. '\n')
+    end
     print(err)
   else
     if #inspections > 0 then
-      print(file .. '\n')
+      if filename then
+        print(filename .. '\n')
+      end
       errors = errors + #inspections
       print(moonpick.format_inspections(inspections))
     end
   end
 end
 
+if opts.stdin then
+  local file = io.stdin:read('*a')
```

Comment:
I suggest a more descriptive name here, such as `content` or `source` instead of `file`.